tests/test_butler.py

```diff
@@ -892,6 +892,25 @@ def testStringification(self):
             for testStr in self.datastoreName:
                 self.assertIn(testStr, datastoreName)
 
+    def testIngestTimeQuery(self):
+        storageClass = self.storageClassFactory.getStorageClass("StructuredDataNoComponents")
+        butler = self.runPutGetTest(storageClass, "test_metric")
+
+        datasets = list(butler.registry.queryDatasets(..., collections=...))
+        len0 = len(datasets)
+        self.assertGreater(len0, 0)
+
+        where = "ingest_date > T'2000-01-01'"
+        datasets = list(butler.registry.queryDatasets(..., collections=..., where=where))
+        len1 = len(datasets)
+        self.assertEqual(len0, len1)
+
+        # no one will ever use this piece of software in 30 years
+        where = "ingest_date > T'2050-01-01'"
+        datasets = list(butler.registry.queryDatasets(..., collections=..., where=where))
+        len2 = len(datasets)
+        self.assertEqual(len2, 0)
+
```

Comment:
Would you mind moving this to `daf/butler/registry/tests/_registry.py`?  You can get all of the set up you need with:
```
registry = self.makeRegistry()
self.loadData(registry, "base.yaml")
self.loadData(registry, "datasets.yaml")
```
That will give us coverage in PostgreSQL, and I've found relying on the outputs of `runPutGetTest` to be a problem in the past because the context managers within it can make it hard to see what actually failed because they swallow exceptions.

Reply:
This happens after runPutGetTest so there should be no issue with swallowing of exceptions.

There is a broader problem that I need to completely refactor test_butler.py so that it moves mostly into the tests module and is clearer to how it tests backends. It mostly is designed to switch datastores but it does of course need to have a testing.postgresql version as well.

Reply:
Thanks for the hint, I'll move it to `_registry.py`. The test grew large, it's hard to navigate their classes and contents to find the right spot to stick something in 🙂